DQM/BeamMonitor/plugins/BeamSpotProblemMonitor.h

```diff
@@ -11,86 +11,71 @@
 #include <string>
 // CMS
 #include "FWCore/Framework/interface/Frameworkfwd.h"
-#include "FWCore/Framework/interface/EDAnalyzer.h"
-#include "FWCore/Framework/interface/Event.h"
 #include "FWCore/ParameterSet/interface/ParameterSet.h"
 #include "DataFormats/Scalers/interface/BeamSpotOnline.h"
 #include "DataFormats/Scalers/interface/DcsStatus.h"
 #include "DataFormats/TrackReco/interface/Track.h"
 #include "DataFormats/TrackReco/interface/TrackFwd.h"
-#include "DQMServices/Core/interface/DQMStore.h"
 #include "DQMServices/Core/interface/MonitorElement.h"
-#include "RecoVertex/BeamSpotProducer/interface/BSTrkParameters.h"
-#include "RecoVertex/BeamSpotProducer/interface/BeamFitter.h"
-#include <fstream>
+#include "DQMServices/Core/interface/DQMEDAnalyzer.h"
 
 
 //
 // class declaration
 //
 
-class BeamSpotProblemMonitor : public edm::EDAnalyzer {
+class BeamSpotProblemMonitor : public DQMEDAnalyzer {
   public:
 
-    BeamSpotProblemMonitor( const edm::ParameterSet& );
-    ~BeamSpotProblemMonitor() override;
+    explicit BeamSpotProblemMonitor( const edm::ParameterSet& );
+    static void fillDescriptions(edm::ConfigurationDescriptions& );
 
   protected:
 
     //The order it runs
 
 
-    // BeginJob
-    void beginJob() override;
-
     // BeginRun
-    void beginRun(const edm::Run& r, const edm::EventSetup& c) override;
+    void bookHistograms(DQMStore::IBooker& i, const edm::Run& r, const edm::EventSetup& c) override;
     void analyze(const edm::Event& e, const edm::EventSetup& c) override;
     void beginLuminosityBlock(const edm::LuminosityBlock& lumiSeg,
         const edm::EventSetup& context) override;
     void endLuminosityBlock(const edm::LuminosityBlock& lumiSeg,
         const edm::EventSetup& c) override;
     // EndRun
     void endRun(const edm::Run& r, const edm::EventSetup& c) override;
-    // Endjob
-    void endJob(const edm::LuminosityBlock& lumiSeg, const edm::EventSetup& c);
```

Comment:
Not being able to add `override` to this function should have been a clue that there was a problem.